### ISSUE TITLE
ERA001 false positives on CODE_INDICATORS

### ISSUE THREAD

**Issue by SXHRYU** [astral-sh/ruff#18461]
### Summary

If the comment consists of one word derived from "return"/"break"/"continue"/"import" (i.e. [CODE_INDICATORS](https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/eradicate/detection.rs#L10-L15)) it is considered valid python code.

Examples:

```python
# returnable  noqa: ERA001
# imports  noqa: ERA001
# breaking  noqa: ERA001
```

It can also be detrimental if it triggers on normal comment that contains function name on a new line:
```python
# this is a long comment explaining a bug that happens when calling
# on_return  noqa: ERA001
# (^ this can be a valid function name)
```

Perhaps it's viable to wrap [`if !CODE_INDICATORS.is_match(line)`](https://github.com/astral-sh/ruff/blob/main/crates/ruff_linter/src/rules/eradicate/detection.rs#L79C1-L81C6) to some less-greedy function?

### Version

0.11.12

**Comment by ntBre** [comment]
My somewhat naive idea would just be to use a `Regex` instead of an `AhoCorasick` for the keywords and throw `\b` on the ends.

It does look like the current implementation is consistent with the [upstream version](https://github.com/PyCQA/eradicate/blob/7ac47683e79ebdb8a86c8f66a24a6a8ced2ea028/eradicate.py#L89-L93), but it would make sense to me to check for an exact match instead.

**Comment by sakshamgoyal-01** [comment]
#25116 
<img width="386" height="63" alt="Image" src="https://github.com/user-attachments/assets/2bbedd12-6489-4173-be23-e62cf29f8c0b" />

https://play.ruff.rs/95b9ac76-aed9-4880-b200-ee7220b586dc)

It will error on tab intentation, but not on spaces.
